cognite/powerops/client/shop/shop_run.py

```diff
@@ -210,6 +211,27 @@ def get_shop_files(self) -> Iterable[str]:
         for shop_file in self._shop_files:
             yield self._download_file(shop_file.external_id)
 
+    def get_pre_run_file(self) -> str:
+        """
+        Get the pre-run file generated by CogSHOP. This file is linked to the SHOP run event through the relationship
+        "relationship_to.pre_run_file" and contains the input data for the SHOP run.
+
+        Returns:
+            The pre-run file as a string.
+        """
+        pre_run_files = self._client.relationships.list(
+            source_external_ids=[self.external_id],
+            source_types=["event"],
+            target_types=["file"],
+            labels=LabelFilter(contains_all=["relationship_to.pre_run_file"]),
+        )
+        if not pre_run_files:
+            raise ValueError("No pre-run file found for SHOP run.")
+        if len(pre_run_files) > 1:
+            raise ValueError("Multiple pre-run files found for SHOP run")

```

Comment:
Could be nice to include the external IDs of the files in the exception, such that it would be easier to debug or fetch one of the files.